src/acl/access-groups.ts

```diff
@@ -0,0 +1,274 @@
+import { IndexedFormula, NamedNode, sym } from 'rdflib'
+import { ACLbyCombination, readACL } from './acl'
+import widgets from '../widgets'
+import ns from '../ns'
+import { AccessController } from './access-controller'
+import { AgentMapMap, ComboList, PartialAgentTriple } from './types'
+import { AddAgentButtons } from './add-agent-buttons'
+
+const ACL = ns.acl
+
+const COLLOQUIAL = {
+  13: 'Owners',
+  9: 'Owners (write locked)',
+  5: 'Editors',
+  3: 'Posters',
+  2: 'Submitters',
+  1: 'Viewers'
+}
+
+const RECOMMENDED = {
+  13: true,
+  5: true,
+  3: true,
+  2: true,
+  1: true
+}
+
+const EXPLANATION = {
+  13: 'can read, write, and control sharing.',
+  9: 'can read and control sharing, currently write-locked.',
+  5: 'can read and change information',
+  3: 'can add new information, and read but not change existing information',
+  2: 'can add new information but not read any',
+  1: 'can read but not change information'
+}
+
+interface AccessGroupsOptions {
+  defaults?: boolean
+}
+
+export class AccessGroups {
+  private readonly defaults: boolean
+  public byCombo: ComboList
+  public aclMap: AgentMapMap
+  private readonly addAgentButton: AddAgentButtons
+  private readonly root: HTMLElement
+
+  constructor (
+    private doc: NamedNode,
+    private aclDoc: NamedNode,
+    public controller: AccessController,
+    public store: IndexedFormula,
+    private options: AccessGroupsOptions = {}
+  ) {
+    this.defaults = options.defaults || false
+    this.aclMap = readACL(doc, aclDoc, store, this.defaults)
+    this.byCombo = ACLbyCombination(this.aclMap)
+    this.addAgentButton = new AddAgentButtons(this)
+    this.root = this.controller.dom.createElement('div')
+    this.root.classList.add(this.controller.classes.accessGroupList)
+  }
+
+  public render (): HTMLElement {
+    this.root.innerHTML = ''
+    this.renderGroups().forEach(group => this.root.appendChild(group))
+    if (this.controller.isEditable) {
+      this.root.appendChild(this.addAgentButton.render())
+    }
+    return this.root
+  }
+
+  private renderGroups (): HTMLElement[] {
```

Comment:
Let's leave the word 'render' for things which make HTML DOM.    Suggest  getGroups here

Reply:
Oh I see it does return DOM elements, it was the names of those  variables were confusing.